tests/app/controllers/test_party_controller.py

```diff
@@ -208,3 +210,12 @@ def test_get_survey_list_details_for_party(self, get_respondent_enrolments, get_
             self.assertTrue(survey_details['survey_short_name'] is not None)
             self.assertTrue(survey_details['business_party_id'] is not None)
             self.assertTrue(survey_details['collection_exercise_ref'] is not None)
+
+    def test_should_not_display_access_button_for_eQ(self):
+        self.assertFalse(should_display_access_button(status='COMPLETE', ci_type='EQ'))
```

Comment:
We could have test fixture-like lists of tuple combinations here instead of three separate tests:

```python
...
    Combination = namedtuple("Combination", ["status", "ci_type", "expected"])
    combinations = [
        Combination('NOTSTARTED', 'SEFT', True),
        Combination('COMPLETE', 'SEFT', False), 
        ...  # etc.
    ]
    for c in combinations:
        self.assertEqual(should_display_access_button(c.status, c.ci_type), c.expected)
    ...
```

Reply:
This is a lot better actually 🌮